### ISSUE TITLE
[BUG] hash_aggregate_test.py::test_exact_percentile_groupby_partial_fallback_to_cpu failed mismatched cpu and gpu output with DATAGEN_SEED=1723492454

### ISSUE THREAD

**Issue by pxLi** [NVIDIA/spark-rapids#11320]
**Describe the bug**
rapids_integration-scala213-dev-github, run: 223

failed in spark 334 integrations tests,
```
AssertionError: GPU and CPU float values are different [0, 'percentile(val, 0.1)']

cpu = -2.6611656653555613e+151, gpu = -2.2241524877840376e+151
float_check = <function get_float_check.<locals>.<lambda> at 0x7fd4e5238c10>
path = [0, 'percentile(val, 0.1)']
```

```
[2024-08-12T22:06:45.648Z] FAILED ../../src/main/python/hash_aggregate_test.py::test_exact_percentile_groupby[[('key', RepeatSeq(Integer)), ('val', Double), ('freq', Long(not_null))]0][DATAGEN_SEED=1723492454, TZ=UTC, IGNORE_ORDER] - AssertionError: GPU and CPU float values are different [0, 'percentile(val, 0.1)']
[2024-08-12T22:06:45.648Z] FAILED ../../src/main/python/hash_aggregate_test.py::test_exact_percentile_groupby_partial_fallback_to_cpu[false-partial-[('key', RepeatSeq(Integer)), ('val', Double), ('freq', Long(not_null))]][DATAGEN_SEED=1723492454, TZ=UTC, IGNORE_ORDER, ALLOW_NON_GPU(ObjectHashAggregateExec,SortAggregateExec,ShuffleExchangeExec,HashPartitioning,AggregateExpression,Alias,Cast,Literal,ProjectExec,Percentile)] - AssertionError: GPU and CPU float values are different [0, 'percentile(val, 0.1)']
[2024-08-12T22:06:45.648Z] FAILED ../../src/main/python/hash_aggregate_test.py::test_exact_percentile_groupby_partial_fallback_to_cpu[false-final|complete-[('key', RepeatSeq(Integer)), ('val', Double), ('freq', Long(not_null))]][DATAGEN_SEED=1723492454, TZ=UTC, IGNORE_ORDER, ALLOW_NON_GPU(ObjectHashAggregateExec,SortAggregateExec,ShuffleExchangeExec,HashPartitioning,AggregateExpression,Alias,Cast,Literal,ProjectExec,Percentile)] - AssertionError: GPU and CPU float values are different [0, 'percentile(val, 0.1)']
[2024-08-12T22:06:45.648Z] FAILED ../../src/main/python/hash_aggregate_test.py::test_exact_percentile_groupby_partial_fallback_to_cpu[true-partial-[('key', RepeatSeq(Integer)), ('val', Double), ('freq', Long(not_null))]][DATAGEN_SEED=1723492454, TZ=UTC, IGNORE_ORDER, ALLOW_NON_GPU(ObjectHashAggregateExec,SortAggregateExec,ShuffleExchangeExec,HashPartitioning,AggregateExpression,Alias,Cast,Literal,ProjectExec,Percentile)] - AssertionError: GPU and CPU float values are different [0, 'percentile(val, 0.1)']
[2024-08-12T22:06:45.648Z] FAILED ../../src/main/python/hash_aggregate_test.py::test_exact_percentile_groupby_partial_fallback_to_cpu[true-final|complete-[('key', RepeatSeq(Integer)), ('val', Double), ('freq', Long(not_null))]][DATAGEN_SEED=1723492454, TZ=UTC, INJECT_OOM, IGNORE_ORDER, ALLOW_NON_GPU(ObjectHashAggregateExec,SortAggregateExec,ShuffleExchangeExec,HashPartitioning,AggregateExpression,Alias,Cast,Literal,ProjectExec,Percentile)] - AssertionError: GPU and CPU float values are different [0, 'percentile(val, 0.1)']
```

```

```

**Steps/Code to reproduce bug**
Please provide a list of steps or a code sample to reproduce the issue.
Avoid posting private or sensitive data.

**Expected behavior**

**Environment details (please complete the following information)**
 - Environment location: Standalone
 - Spark configuration settings related to the issue

**Additional context**

**Comment by revans2** [comment]
This is caused by https://issues.apache.org/jira/browse/SPARK-45599 it was fixed in Spark 4.0.0 and Spark 3.5.2. I looked closer at the test results and I saw that there was a -0.0 and a regular 0.0 as one of the percentiles produced.

It is also a dupe of #10233 (I wish the github issue search was better)

I think we probably want to have the tests avoid outputting both -0.0 and 0.0 for versions that do not have the fix in it.
